lib/sshutils/ctx.go

```diff
@@ -143,6 +147,30 @@ func (c *ConnectionContext) GetForwardAgent() bool {
 	return c.forwardAgent
 }
 
+// TryIncrSessions tries to increment the active session count; if ok the
+// returned decr function *must* be called when the associated session is closed.
+func (c *ConnectionContext) IncrSessions(max int64) (decr func(), ok bool) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.sessions >= max {
+		return nil, false
```

Comment:
to make this fool-proof
```suggestion
		return func() {}, false
```